File: C# (ASP.NET & WPF)/C# - INSY7315 - EliteRentals - (Web App, Mobile App & API)/EliteRentals-WebApp/Models/DTOs/LeaseDto.cs
namespace EliteRentals.Models.DTOs
{
    public class LeaseDto
    {
        public int LeaseId { get; set; }
        public int PropertyId { get; set; }
        public int TenantId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal Deposit { get; set; }
        public string Status { get; set; } = "Active";
        public byte[]? DocumentData { get; set; }
        public string? DocumentType { get; set; }

        public PropertyDto Property { get; set; }
        public TenantDto Tenant { get; set; }

        // convenience display helpers (for views)
        public string TenantName => $"{Tenant?.FirstName} {Tenant?.LastName}";
        public string PropertyTitle => Property?.Title ?? "Unknown Property";

        public DateTime? ArchivedDate { get; set; }
    }

    public class LeaseCreateUpdateDto
    {
        public int LeaseId { get; set; }
        public int PropertyId { get; set; }
        public int TenantId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal Deposit { get; set; } // nullable
        public string Status { get; set; }
    }

    public class PropertyDto
    {
        public int PropertyId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public decimal RentAmount { get; set; }
        public string Status { get; set; }
    }

    public class TenantDto
    {
        public int UserId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string Role { get; set; }
    }
}

File: C# (ASP.NET & WPF)/C# - INSY7315 - EliteRentals - (Web App, Mobile App & API)/Elite-Rentals-Mobile-App/app/src/main/java/com/rentals/eliterentals/LeaseDto.kt
package com.rentals.eliterentals
data class LeaseDto(
    val leaseId: Int,
    val propertyId: Int,
    val property: PropertyDto?, // <-- must match API
    val tenantId: Int,
    val tenant: TenantDto?,     // <-- must match API
    val startDate: String,
    val endDate: String,
    val deposit: Double,
    val status: String,
    val documentData: String? = null,
    val documentType: String? = null
)


data class TenantDto(
    val userId: Int,
    val firstName: String,
    val lastName: String,
    val email: String
)


